packages/typespec-client-generator-core/src/methods.ts

```diff
@@ -595,23 +595,20 @@ function getSdkMethodResponse(
     } else if (responseTypes.size === 1) {
       type = allResponseBodies[0];
     }
-    if (nonBodyExists && type) {
-      type = {
-        kind: "nullable",
-        name: createGeneratedName(context, operation, "NullableResponse"),
-        crossLanguageDefinitionId: `${getCrossLanguageDefinitionId(context, operation)}.NullableResponse`,
-        isGeneratedName: true,
-        type: type,
-        decorators: [],
-        access: "public",
-        usage: UsageFlags.Output,
-        namespace: client.namespace,
-      };
-    }
   }
+  
+  // Set optional property based on whether responses have bodies
+  let optional: boolean | undefined = undefined;
+  if (type !== undefined) {
+    // If we have a response type, set optional based on whether some responses lack bodies
+    optional = nonBodyExists;
```

Comment:
nit, name of `nonBodyExists` could be confusing. Maybe `opContainsResponseWithoutBody`?